src/odemis/driver/simcam.py

```diff
@@ -109,7 +109,7 @@ def clip_max_res(img_res):
         self.binning = model.ResolutionVA(self._binning,
                               ((1, 1), (16, 16)), setter=self._setBinning)
 
-        hlf_shape = (self._shape[0] // 2 - 1, self._shape[1] // 2 - 1)
+        hlf_shape = (self._img.shape[-1] // 2 - 1, self._img.shape[-2] // 2 - 1)
```

Comment:
I think this works differently than the previous code for 2D images? I'm not sure if these images are ever 2d, but comment above indicates that grayscale is supported?

```suggestion
     dim0, dim1 = 0, 1
     if self._img.ndim == 3:
         dim0, dim1 = -1, -2
     hlf_shape = (self._img.shape[dim0] // 2 - 1, self._img.shape[dim1] // 2 - 1)
```

Reply:
Typically the images are greyscale (last two dimensions are YX), but it has to also handle RGB images, of dimensions YXC, so the change indeed doesn't work. However the proposed change by Patrick also doesn't work, as it would mix up the dimensions, in both cases.
I think just writing img.shape[1] ..... img.shape[0] would work. You'd want to add a comment to indicate that the numpy dimension order is YX(C).